presto-docs/src/main/sphinx/connector/elasticsearch.rst

```diff
@@ -0,0 +1,171 @@
+
+``elasticsearch.scroll-size``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This property defines the maximum number of hits to be returned with each
+batch of Elasticsearch scroll.
+
+This property is optional; the default is ``1000``.
+
+``elasticsearch.scroll-time``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+
+This property defines number of milliseconds Elasticsearch client would
+keep live for a search scroll result.
+
+This property is optional; the default is ``20s``.
+
+``elasticsearch.max-hits``
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This property defines the maximum number of hits an Elasticsearch request can fetch.
+
+This property is optional; the default is ``1000000``.
+
+``elasticsearch.request-timeout``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This property defines timeout for all Elasticsearch request.
+
+This property is optional; the default is ``10s``.
+
+Table Definition Files
+----------------------
+
+For Presto, Elasticsearch data must be mapped to tables and columns to allow queries against the data.
+
+A table definition file consists of a JSON definition for a table. The
+name of the file can be arbitrary, but it must end with ``.json``.
+
+.. code-block:: none
+
+    {
+        "tableName": ...,
+        "schemaName": ...,
+        "hostAddress": ...,
+        "port": ...,
+        "clusterName": ...,
+        "index": ...,
+        "indexExactMatch": ...,
+        "type": ...
+        "columns": [
+            {
+                "name": ...,
+                "type": ...,
+                "jsonPath": ...,
+                "jsonType": ...,
+                "ordinalPosition": ...
+            }
+        ]
+    }
+
+=================== ========= ============== =============================
+Field               Required  Type           Description
+=================== ========= ============== =============================
+``tableName``       required  string         Presto table name defined by this file.
+``schemaName``      optional  string         Schema that contains the table. If omitted, the default schema name is used.
+``host``            required  string         Elasticsearch search node host name.
+``port``            required  integer        Elasticsearch search node port number.
+``clusterName``     required  string         Elasticsearch cluster name.
+``index``           required  string         Elasticsearch index that is backing this table.
+``indexExactMatch`` optional  boolean        true requires index name exact match, false indicates sending request to all indices with the same prefix.
+``type``            required  string         Elasticsearch type, which represents a group of similar documents.
+``columns``         optional  list           List of column metadata information.
+=================== ========= ============== =============================
+
+Elasticsearch Column Metadata
+-----------------------------
+
+Optionally, Elasticsearch column metadata can be configured in json file, with the following fields:
```

Comment:
Why is this optional? If the user doesn't describe the columns how will the connector behave?